web/src/lib/components/asset-viewer/photo-editor.svelte

```diff
@@ -0,0 +1,629 @@
+<script lang="ts">
+  import ImageEditor from 'tui-image-editor?client';
+  import { onMount } from 'svelte';
+  import { browser } from '$app/environment';
+  import { api, AssetResponseDto } from '@api';
+  import { handleError } from '$lib/utils/handle-error';
+  import FullScreenModal from '../shared-components/full-screen-modal.svelte';
+  import Button from '../elements/buttons/button.svelte';
+  import AutoFix from 'svelte-material-icons/AutoFix.svelte';
+  import ImageFilterHdr from 'svelte-material-icons/ImageFilterHdr.svelte';
+  import WeatherSunny from 'svelte-material-icons/WeatherSunny.svelte';
+  import WeatherCloudy from 'svelte-material-icons/WeatherCloudy.svelte';
+  import CropRotate from 'svelte-material-icons/CropRotate.svelte';
+  import Tune from 'svelte-material-icons/Tune.svelte';
+  import ImageAutoAdjust from 'svelte-material-icons/ImageAutoAdjust.svelte';
+  import Fullscreen from 'svelte-material-icons/Fullscreen.svelte';
+  import RelativeScale from 'svelte-material-icons/RelativeScale.svelte';
+  import RectangleOutline from 'svelte-material-icons/RectangleOutline.svelte';
+  import SquareOutline from 'svelte-material-icons/SquareOutline.svelte';
+  import Brightness6 from 'svelte-material-icons/Brightness6.svelte';
+  import Close from 'svelte-material-icons/Close.svelte';
+  import DotsVertical from 'svelte-material-icons/DotsVertical.svelte';
+  import FlipHorizontal from 'svelte-material-icons/FlipHorizontal.svelte';
+  import FlipVertical from 'svelte-material-icons/FlipVertical.svelte';
+  import FormatRotate90 from 'svelte-material-icons/FormatRotate90.svelte';
+  import TriangleSmallUp from 'svelte-material-icons/TriangleSmallUp.svelte';
+
+  import SuggestionsButton from './photo-editor/SuggestionsButton.svelte';
+  import AspectRatioButton from './photo-editor/aspect-ratio-button.svelte';
+  import AdjustElement from './photo-editor/adjust-element.svelte';
+  import FilterCard from './photo-editor/filter-card.svelte';
+
+  import { createEventDispatcher } from 'svelte';
+
+  const dispatch = createEventDispatcher();
+
+  let editorElement: HTMLDivElement;
+
+  let angle = 0;
+  let angleSlider: HTMLElement;
+  let angleSliderHandle: HTMLElement;
+
+  let activeButton: 'autofix' | 'crop' | 'adjust' | 'filter' = 'autofix';
+  type activeEdit = 'optimized' | 'dynamic' | 'warm' | 'cold' | '';
+  let activeEdit: activeEdit;
+
+  type aspectRatio =
+    | 'free'
+    | 'square'
+    | 'original'
+    | '16_9'
+    | '9_16'
+    | '5_4'
+    | '4_5'
+    | '4_3'
+    | '3_4'
+    | '3_2'
+    | '2_3'
+    | 'square';
+  let aspectRatio: aspectRatio = 'free';
+
+  export let asset: AssetResponseDto;
+  let assetData: string;
+  let publicSharedKey = '';
+  let imageEditor: ImageEditor;
+
+  onMount(async () => {
+    let maxWidth: string = String(window.innerWidth);
```

Comment:
Maybe use `window.innerWidth?.toString()` instead? `String()` will call this eventually under the hood anyway and this will avoid the case of `String(null) === "null"`.